### ISSUE TITLE
ModuleNotFoundError: No module named 'joblib.my_exceptions'

### ISSUE THREAD

**Issue by billylaing** [civisanalytics/civis-python#468]
With the latest release of [joblib@1.3.1](https://pypi.org/project/joblib/#history) the "my_exceptions" module was deprecated/removed. 

```
Traceback (most recent call last):
  File "/Users/blaing/Workspace/avl/archive-tables/src/archive_tables.py", line 4, in <module>
    from civis.io import civis_to_csv, query_civis
  File "/Users/blaing/.local/share/virtualenvs/archive-tables-P6xH7wLS/lib/python3.10/site-packages/civis/__init__.py", line 4, in <module>
    from civis import io, ml, parallel, utils
  File "/Users/blaing/.local/share/virtualenvs/archive-tables-P6xH7wLS/lib/python3.10/site-packages/civis/parallel.py", line 17, in <module>
    from joblib.my_exceptions import TransportableException
ModuleNotFoundError: No module named 'joblib.my_exceptions'
```

This library is required by `civis@1.16.0` but will break if `joblib` > 1.2 is installed.

https://github.com/civisanalytics/civis-python/blob/34ed2de06f1151f5d4e10af121007efdd04178bf/requirements.txt#L6

**Comment by billylaing** [comment]
Thanks. Y'all would probably know best based on usage and numbers of users installing, but this seems like a really good opportunity for a patch. 

I'd be happy to make a PR to patch `requirements.txt` until the more extensive updates for joblib changes lands. Thoughts?

**Comment by billylaing** [comment]
Ah, I see, someone has already created one:

https://github.com/civisanalytics/civis-python/pull/466

**Comment by jacksonlee-civis** [comment]
We've just released [civis 1.16.1 on PyPI](https://pypi.org/project/civis/1.16.1/) with joblib pinned at < 1.3.0, though I think we can leave this issue open since the 1.16.1 release was directly based off of [1.16.0 from Dec 2021](96a31a77fcf7c9678052f55aafe2939e9f56874f) (we've started checking in API-breaking changes since 1.16.0). On the `main` branch right now (at 34ed2de06f1151f5d4e10af121007efdd04178bf), this joblib issue is still outstanding.

**Comment by mattyb** [comment]
Resolved by #467

**Comment by jacksonlee-civis** [comment]
Update: civis-python v2.3.0 is now available, where joblib is pinned at `>= 1.3.0`.
